core/templates/dev/head/domain/learner_dashboard/LearnerDashboardActivityIdsObjectFactory.js

```diff
@@ -0,0 +1,85 @@
+// Copyright 2015 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Factory for creating new frontend instances of learner
+   dashboard activity ids domain object.
+ */
+
+oppia.factory('LearnerDashboardActivityIdsObjectFactory', [function() {
+  var LearnerDashboardActivityIds = function(
+    incompleteExplorationIds, incompleteCollectionIds,
+    completedExplorationIds, completedCollectionIds, explorationPlaylistIds,
+    collectionPlaylistIds) {
+    this.incompleteExplorationIds = incompleteExplorationIds;
+    this.incompleteCollectionIds = incompleteCollectionIds;
+    this.completedExplorationIds = completedExplorationIds;
+    this.completedCollectionIds = completedCollectionIds;
+    this.explorationPlaylistIds = explorationPlaylistIds;
+    this.collectionPlaylistIds = collectionPlaylistIds;
+  };
+
+  LearnerDashboardActivityIds.prototype.includesActivity = (
+    function(activityId) {
+      if (this.incompleteCollectionIds.indexOf(activityId) !== -1 ||
+          this.completedCollectionIds.indexOf(activityId) !== -1 ||
+          this.collectionPlaylistIds.indexOf(activityId) !== -1 ||
+          this.incompleteExplorationIds.indexOf(activityId) !== -1 ||
+          this.completedExplorationIds.indexOf(activityId) !== -1 ||
+          this.explorationPlaylistIds.indexOf(activityId) !== -1) {
+        return true;
+      } else {
+        false;
```

Comment:
Careless; no return value. Why is this branch of the code not tested?

Reply:
Done! Added tests.